src/test/java/teammates/test/cases/ui/browsertests/InstructorHomePageUiTest.java

```diff
@@ -296,8 +296,45 @@ public void testRemindActions(){
 
     }
 
-    public void testPublishUnpublishActions(){
-        //TODO add test for publishing and unpublishing feedback sessions
+    public void testPublishUnpublishActions() {
+        ______TS("publish action: AWAITING feedback session");
+        
+        homePage.verifyUnclickable(homePage.getPublishLink(feedbackSession_AWAITING.courseId, feedbackSession_AWAITING.feedbackSessionName));
+
+        ______TS("publish action: OPEN feedback session");
+        
+        homePage.clickFeedbackSessionPublishLink(feedbackSession_OPEN.courseId, feedbackSession_OPEN.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_PUBLISHED);
+
+        ______TS("unpublish action: OPEN feedback session");
+        
+        homePage.clickFeedbackSessionUnpublishLink(feedbackSession_OPEN.courseId, feedbackSession_OPEN.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+
+        ______TS("publish action: CLOSED feedback session");
+        
+        homePage.clickFeedbackSessionPublishLink(feedbackSession_CLOSED.courseId, feedbackSession_CLOSED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_PUBLISHED);
+
+        ______TS("unpublish action: CLOSED feedback session");
+        
+        homePage.clickFeedbackSessionUnpublishLink(feedbackSession_CLOSED.courseId, feedbackSession_CLOSED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+        
+        ______TS("unpublish action: PUBLISHED feedback session");
+        homePage.clickFeedbackSessionUnpublishLink(feedbackSession_PUBLISHED.courseId, feedbackSession_PUBLISHED.feedbackSessionName);
+        homePage.waitForPageToLoad();
+        homePage.verifyStatus(Const.StatusMessages.FEEDBACK_SESSION_UNPUBLISHED);
+        homePage.goToPreviousPage(InstructorHomePage.class);
```

Comment:
Is this needed? I thought the instructor remains on the same page after clicking publish or unpublish now.